pkg/minikube/machine/fix.go

```diff
@@ -158,25 +159,21 @@ func maybeWarnAboutEvalEnv(drver string, name string) {
 	if !driver.IsKIC(drver) {
 		return
 	}
+	sh, err := shell.Detect()
+	ec := shell.EnvConfig{Shell: sh}
+	if err != nil {
+		out.ErrT(out.Notice, "Cannot discover shell type. Fallback to {{.shell}} which may not be accurate", out.V{"shell": shell.DefaultShellName})
+	}
 	if os.Getenv(constants.MinikubeActiveDockerdEnv) != "" {
 		out.T(out.Notice, "Noticed you have an activated docker-env on {{.driver_name}} driver in this terminal:", out.V{"driver_name": drver})
-		// TODO: refactor docker-env package to generate only eval command per shell. https://github.com/kubernetes/minikube/issues/6887
-		out.WarningT(`Please re-eval your docker-env, To ensure your environment variables have updated ports:
-
-	'minikube -p {{.profile_name}} docker-env'
-
-	`, out.V{"profile_name": name})
+		msg := shell.GenerateUsageHint(ec, "eval docker-env", "minikube -p {{.profile_name}} docker-env")
```

Comment:
there is no need to  eval docker-env, 
I suggest this function take two parameters
1- shell type (bash, powershell,fish,...)
2-profile name

and based on that it give you the exact command to eval.
that way the code will look cleaner.

Reply:
Hey,
I'm sorry but I'm missing sth here. `eval docker-env,` is just usage comment. I can replace it with something more descriptive like `please eval docker-env using given snippet` or similar. Variable `ec`  contains shell type.

Do you want me to add new function to https://github.com/kubernetes/minikube/blob/master/pkg/minikube/shell/shell.go#L86, so string `"minikube -p {{.profile_name}} docker-env"` will be extracted to shell pkg and invocation will look like: 
```golang
msg := shell.GenerateDockerEnvEval(ec, profile) 
out.WarningT(msg)
```
This will require adding 2 new functions to shell pkg (1 for podman 1 for docker).

Can I ask for additional clarification please?

Reply:
`2567e3af4a8ee655e47dac63831f54f5107457f0` provides some fixes here. Not sure if it's exactly what was requested.